src/Microsoft.Dnx.Runtime/Helpers/RoslynAnalyzerHelper.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.Dnx.Compilation;
+using NuGet;
+
+namespace Microsoft.Dnx.Runtime.Helpers
+{
+    internal static class RoslynAnalyzerHelper
+    {
+        private static readonly string AnalyzerBaseFolder = "analyzers";
+        private static readonly string AnalyzerLanguage = "cs";
+        private static readonly string AnalyzerExtension = ".dll";
+
+        public static IEnumerable<IAnalyzerReference> GetAnalyerReferences(PackageInfo packageInfo,
+                                                                           string packagePath,
+                                                                           FrameworkName runtimeFramework)
+        {
+            // assumption: analyzers' paths are in following format
+            // analyzer/{framework}/{cs|vb}/{filename}.dll
+            var files = packageInfo.LockFileLibrary.Files.Select(CreateFileReference)
```

Comment:
Can we do the filtering of files before the Select so we can get rid of the "Where-not-null" check?